compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java

```diff
@@ -172,41 +177,15 @@ private PackageManifest parseAsPackageManifest() {
                 authors = getStringArrayFromPackageNode(pkgNode, AUTHORS);
                 keywords = getStringArrayFromPackageNode(pkgNode, KEYWORDS);
                 exported = getStringArrayFromPackageNode(pkgNode, EXPORT);
+                includes = getStringArrayFromPackageNode(pkgNode, INCLUDES);
                 repository = getStringValueFromTomlTableNode(pkgNode, REPOSITORY, "");
-                ballerinaVersion = getStringValueFromTomlTableNode(pkgNode, "distribution", "");
-                visibility = getStringValueFromTomlTableNode(pkgNode, "visibility", "");
+                ballerinaVersion = getStringValueFromTomlTableNode(pkgNode, DISTRIBUTION, "");
+                visibility = getStringValueFromTomlTableNode(pkgNode, VISIBILITY, "");
                 template = getBooleanFromTemplateNode(pkgNode, TEMPLATE);
-                icon = getStringValueFromTomlTableNode(pkgNode, "icon", "");
-
-                // validate icon path for only png files
-                // we ignore other file types here, since file type error will be shown
-                if (icon != null && hasPngExtension(icon)) {
-                    Path iconPath = Paths.get(icon);
-                    if (!iconPath.isAbsolute()) {
-                        iconPath = this.projectPath.resolve(iconPath);
-                    }
+                icon = getStringValueFromTomlTableNode(pkgNode, ICON, "");
 
-                    if (Files.notExists(iconPath)) {
-                        // validate icon path
-                        // if file path does not exist, throw this error
-                        reportDiagnostic(pkgNode.entries().get("icon"),
-                                "could not locate icon path '" + icon + "'",
-                                "error.invalid.path", DiagnosticSeverity.ERROR);
-                    } else {
-                        // validate file content
-                        // if other file types renamed as png, throw this error
-                        try {
-                            if (!FileUtils.isValidPng(iconPath)) {
-                                reportDiagnostic(pkgNode.entries().get("icon"),
-                                        "invalid 'icon' under [package]: 'icon' can only have 'png' images",
-                                        "error.invalid.icon", DiagnosticSeverity.ERROR);
-                            }
-                        } catch (IOException e) {
-                            // should not reach to this line
-                            throw new ProjectException("failed to read icon: '" + icon + "'");
-                        }
-                    }
-                }
+                // we ignore file types except png here, since file type error will be shown
+                validateIconPathForPng(icon, pkgNode);
```

Comment:
We should also validate the includes paths and report an error diagnostic similar to the validation done to the `icon` field

Reply:
Will do that. 

Does adding path validation here eliminate the requirement for path checking in `BalaWriter`?

If we have to validate the path in two places can you suggest a place to move the validation code that can be used by `BalaWriter` and `ManifestBuilder`?